src/io/flutter/server/vmService/VMServiceManager.java

```diff
@@ -245,6 +248,16 @@ private void addServiceExtension(String name) {
       else if (!stream.getValue()) {
         stream.setValue(true);
       }
+
+      if (serviceExtensionState.get(name) != null && serviceExtensionState.get(name).getValue()) {
```

Comment:
add a 
getServiceExtensionCurrentValue method that returns the value.
In that method you should synchronize on 
serviceExtensionsState as that is used to keep all additions to that map safe.

Reply:
Used the pre-existing getServiceExtensionState() method and called .getValue() on it. This should be fine as getServiceExtensionState() is synchronized on serviceExtensionState.